test/powershell/Language/Operators/NullConditional.Tests.ps1

```diff
@@ -0,0 +1,210 @@
+
+        if ($skipTest) {
+            Write-Verbose "Test Suite Skipped. The test suite requires the experimental feature 'PSNullCoalescingOperators' to be enabled." -Verbose
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+        } else {
+            $someGuid = New-Guid
+            $typesTests = @(
+                @{ name = 'string'; valueToSet = 'hello' }
+                @{ name = 'dotnetType'; valueToSet = $someGuid }
+                @{ name = 'byte'; valueToSet = [byte]0x94 }
+                @{ name = 'intArray'; valueToSet = 1..2 }
+                @{ name = 'stringArray'; valueToSet = 'a'..'c' }
+                @{ name = 'emptyArray'; valueToSet = @(1, 2, 3) }
+            )
+        }
+    }
+
+    AfterAll {
+        if ($skipTest) {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        }
+    }
+
+    Context "Null conditional assignment operator ??=" {
+        It 'Variable doesnot exist' {
+
+            Remove-Variable variableDoesNotExist -ErrorAction SilentlyContinue -Force
+
+            $variableDoesNotExist ??= 1
+            $variableDoesNotExist | Should -Be 1
+
+            $variableDoesNotExist ??= 2
+            $variableDoesNotExist | Should -Be 1
+        }
+
+        It 'Variable exists and is null' {
+            $variableDoesNotExist = $null
+
+            $variableDoesNotExist ??= 2
+            $variableDoesNotExist | Should -Be 2
+        }
+
+        It 'Validate types - <name> can be set' -TestCases $typesTests {
+            param ($name, $valueToSet)
+
+            $x = $null
+            $x ??= $valueToSet
+            $x | Should -Be $valueToSet
+        }
+
+        It 'Validate hashtable can be set' {
+            $x = $null
+            $x ??= @{ 1 = '1' }
+            $x.Keys | Should -Be @(1)
+        }
+
+        It 'Validate lhs is returned' {
+            $x = 100
+            $x ??= 200
+            $x | Should -Be 100
+        }
+
+        It 'Rhs is a cmdlet' {
+            $x ??= (Get-Alias -Name 'where')
+            $x.Definition | Should -BeExactly 'Where-Object'
+        }
+
+        It 'Lhs is DBNull' {
+            $x = [System.DBNull]::Value
+            $x ??= 200
+            $x | Should -Be 200
+        }
+
+        It 'Lhs is AutomationNull' {
+            $x = [System.Management.Automation.Internal.AutomationNull]::Value
+            $x ??= 200
+            $x | Should -Be 200
+        }
+
+        It 'Lhs is NullString' {
+            $x = [NullString]::Value
+            $x ??= 200
+            $x | Should -Be 200
+        }
+
+        It 'Error case' {
+            $e = $null
+            $null = [System.Management.Automation.Language.Parser]::ParseInput('1 ??= 100', [ref] $null, [ref] $e)
+            $e[0].ErrorId | Should -BeExactly 'InvalidLeftHandSide'
+        }
+
+        It 'Variable is non-null' {
```

Comment:
Might be also worth testing some falsey but non-null cases (like `''`)

Reply:
Added test